tests/components/mystrom/test_config_flow.py

```diff
@@ -0,0 +1,130 @@
+
+
+async def test_form_duplicates(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, config_entry: MockConfigEntry
+) -> None:
+    """Test abort on duplicate."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch(
+        "pymystrom.get_device_info",
+        return_value={"type": 101, "mac": DEVICE_MAC},
+    ) as mock_session:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] == FlowResultType.ABORT
+        assert result2["reason"] == "already_configured"
+
+    assert len(mock_session.mock_calls) == 1
+
+
+async def test_step_import(hass: HomeAssistant) -> None:
+    """Test the import step."""
+    conf = {
+        CONF_HOST: "1.1.1.1",
+    }
+    with patch("pymystrom.switch.MyStromSwitch.get_state"), patch(
+        "pymystrom.get_device_info",
+        return_value={"type": 101, "mac": DEVICE_MAC},
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=conf
+        )
+        assert result["type"] == FlowResultType.CREATE_ENTRY
+        assert result["title"] == "myStrom Device"
+        assert result["data"] == {
+            CONF_HOST: "1.1.1.1",
+        }
+
+
+async def test_wong_answer_from_device(hass: HomeAssistant) -> None:
+    """Test handling of wrong answers from the device."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+    with patch(
+        "pymystrom.get_device_info",
+        side_effect=MyStromConnectionError(),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] == FlowResultType.FORM
+        assert result2["step_id"] == "user"
+        assert result2["errors"] == {"base": "cannot_connect"}
+    with patch(
+        "pymystrom.get_device_info",
+        return_value={"type": 101, "mac": DEVICE_MAC},
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+            },
+        )
+        assert result2["title"] == "myStrom Device"
+        assert result2["data"] == {"host": "1.1.1.1"}
+        assert result2["type"] == FlowResultType.CREATE_ENTRY
```

Comment:
Missing `block_till_done`
```suggestion
        await hass.async_block_till_done()
    assert result2["type"] == FlowResultType.CREATE_ENTRY
    assert result2["title"] == "myStrom Device"
    assert result2["data"] == {"host": "1.1.1.1"}
```